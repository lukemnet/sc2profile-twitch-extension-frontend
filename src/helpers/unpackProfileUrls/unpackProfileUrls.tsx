import { StarCraft2API } from 'starcraft2-api';

const unpackProfileUrls = (urls: string[], includeLocale = false) =>
  urls.map((profileUrl: string) =>
    StarCraft2API.unpackProfileUrl(profileUrl, includeLocale));

export default unpackProfileUrls;
