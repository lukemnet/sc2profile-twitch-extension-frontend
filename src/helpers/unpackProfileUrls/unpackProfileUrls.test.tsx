import unpackProfileUrls from './unpackProfileUrls';

const urls = [
  'https://starcraft2.com/en-us/profile/1/1/1',
  'https://starcraft2.com/en-us/profile/2/1/2',
  'https://starcraft2.com/en-us/profile/3/1/3',
]

it('unpacks profile URLs correctly', () => {
  expect(unpackProfileUrls(urls)).toMatchSnapshot();
});