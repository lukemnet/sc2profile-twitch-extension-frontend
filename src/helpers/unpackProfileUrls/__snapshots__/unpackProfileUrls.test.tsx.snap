// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`unpacks profile URLs correctly 1`] = `
Array [
  Object {
    "profileId": "1",
    "realmId": "1",
    "regionId": "1",
  },
  Object {
    "profileId": "1",
    "realmId": "1",
    "regionId": "2",
  },
  Object {
    "profileId": "1",
    "realmId": "1",
    "regionId": "3",
  },
]
`;
